### ISSUE TITLE
Board Archive Feature Request

### ISSUE THREAD

**Issue by KurobaExReports** [KurobaExReports/Reports#206]
When I click on a thread in the board archive and I click the back button, it currently takes me back to the board catalog. What it should do instead is take me back to the board archive view/what I searched on the board archive. Right now if I search for something on the board archive and I click on a thread then hit the back button, I have to search for that same thing again to read the other threads. This is very cumbersome

**Comment by K1rakishou** [comment]
Do you mean local search or global search?

**Comment by KurobaExReports** [comment]
Local I guess? What I mean is when I select a board, then I click the three dots on the top right, then I click "View board archive", the search button I am talking about is on that page. 

What I want is when I click on a thread archived on that page, and then I go back, I want to end up back on the "board archives" page. Currently this is not happening. Right now when I click on a thread on the "Board archives" page and then I click back, it's taking me back to the board catalog. This is bad because I have to click those three dots, then click "View board archive" if I want to keep browsing board archives. When I click back I expect to end up back on the "Board archives" page, not the catalog

**Comment by K1rakishou** [comment]
Yeah, this will not be easy to implement due to have navigation works in this app. I will have to rewrite lots of stuff to do that properly.

**Comment by K1rakishou** [comment]
This thing works exactly as you want it to in KurobaExLite so I suggest you to try it out.
